File: backend-api/src/main/java/br/com/sample/api/service/BookService.java
package br.com.sample.api.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import br.com.sample.api.model.BookModel;
import br.com.sample.api.persistence.BookMapper;

public class BookService {

	private BookMapper bookMapper;

	public BookService(BookMapper bookMapper) {
		this.bookMapper = bookMapper;
	}

	public List<BookModel> get(){
		return bookMapper.getAll();
	}
	
	public Long save(BookModel book){
		return bookMapper.save(book);
	}	
	
	public void delete(Long id){
		bookMapper.delete(id);
	}	
	
	public void update(Long id, BookModel book){
		book.setId(id);
		bookMapper.update(book);
	}	
	
	public BookModel getOne(Long id){
		return bookMapper.getOne(id);
	}
	
}

File: frontend/src/app/books/BookService.js
(function(){
  'use strict'

  angular
    .module('bookStoreApp')
    .service('BookService', BookService);

  BookService.$inject = ['$http'];

  function BookService($http){

    return {
      getAll : getAll,
      save : save,
      deleteById : deleteById,
      update : update,
      find : find
    }

    function getAll(){
      return $http.get("http://localhost:8080/api/books")
          .then(function(response){
            return response.data;
          })
          .catch(function(error){
            console.log(error.data);
          });
    };

    function save(book){
      return $http.post('http://localhost:8080/api/books', book)
          .then(function(response){
            return response.data;
          })
          .catch(function(error){
            console.log(error.data);
          });
    }

    function deleteById(id){
      return $http.delete('http://localhost:8080/api/books/' + id)
          .then(function(response){
            return response.data;
          })
          .catch(function(error){
            console.log(error.data);
          });
    }

    function update(book){
      return $http.put('http://localhost:8080/api/books/'+ book.id, book)
          .then(function(response){
            return response.data;
          })
          .catch(function(error){
            console.log(error.data);
          });
    }

    function find(id){
      return $http.get("http://localhost:8080/api/books/"+id)
        .then(function(response){
          return response.data;
        })
        .catch(function(error){
          console.log(error.data);
        });
    }

  }
})();
